### PR TITLE
Unify and simplify train-extract-cache-plot pipeline

### PR DESCRIPTION
**Pull request by jvrsgsty**
This PR aims to clean up the current training pipeline by:
* Providing a single custom optimizer to store custom buffers at any step in training
* Providing a single weight and buffer extraction script, agnostic to the names of the buffers saved in the checkpoints
* Providing a single caching script for visualizations, with (maybe) basic plotting functionality. The idea is to generate plotting caches with relevant metric, that can then be loaded by a custom plotting function (maybe even in a jupyter notebook) to fine tune visualizations.

### CODE REVIEW

**Comment by jvrsgsty** [comment]
Note that as per https://github.com/pytorch/pytorch/issues/1489 we will not have access to named tensors inside a torch optimizer. We'll still rely on the parameters in the optimizer state appearing in the same order as the model parameters.

**Comment by jvrsgsty** [comment]
Still need to test on larger model. For now, translation and performance work great on logistic regression. All other plots filter only conv layers.
